tests/integration/helpers.py

```diff
@@ -0,0 +1,47 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Helpers for the glauth integration tests."""
+
+import logging
+import pathlib
+import zipfile
+from typing import Dict
+from urllib import request
+
+import toml
+
+logger = logging.getLogger(__name__)
+
+CONFIG = "sample-simple.cfg"
+CONFIG_URL = (
+    f"https://github.com/glauth/glauth/raw/0e7769ff841e096dbf0cb67768cbd2ab7142f6fb/v2/{CONFIG}"
+)
+ZIP = "config.zip"
+
+
+def _update_config() -> None:
+    """Update config for integration test."""
+    assert pathlib.Path(CONFIG).exists()
+    with open(CONFIG, "r") as f:
+        config_dict = toml.load(f)
+    config_dict["ldaps"].update({"listen": "0.0.0.0:636", "enabled": True})
+    config_dict["backend"].update({"anonymousdse": True})
+    config_dict["users"][1].update({"homeDir": "/"})
+
+    with open(CONFIG, "w") as f:
+        _ = toml.dump(config_dict, f)
+
+    with zipfile.ZipFile(ZIP, "w") as z:
+        z.write("sample-simple.cfg")
```

Comment:
You do not actually need to write your modified toml configuration file to the disk here. You can just use the _zipfile.ZipFile.writestr_ method instead:

```suggestion
    with zipfile.ZipFile(ZIP, "w") as z:
        z.writestr("sample-simple.cfg", toml.dumps(CONFIG))
```

Also, side note, I discovered that `urllib.request.urlretrieve` may be deprecated sometime in the future: https://docs.python.org/3/library/urllib.request.html#legacy-interface. Not something I think we need to be worried about right now, but something we should probably be aware of.